pkg/analyzer/detection/detection.go

```diff
@@ -15,15 +16,27 @@ import (
 // LogLayerName is the name of the log field holding the detection target.
 const LogLayerName = "layer"
 
+// DetectComponentOpts contains configurations how components detection works
+type DetectComponentOpts struct {
+	// UncertifiedRHEL is boolean to decide if OS is uncertified RHEL
+	UncertifiedRHEL bool
+
+	// IsRHCOSRequired: if CoreOS is required for DetectComponents, Node scanning is disabled if false
+	IsRHCOSRequired bool
```

Comment:
I would be a bit more explicit here. My suggestion:

```suggestion
	// StopEarlyForNonRHCOS: if true, the analysis will run only for namespace starting with `rhcos:`. If false, the analysis will stop after detecting the namespace.
	StopEarlyForNonRHCOS bool
```

Reply:
Nit: I found the naming of `IsRHCOSRequired` a bit confusing as well. 
If I read this as consumer of the lib, I would ask myself "in which case would RHCOS be required?".

@vikin91's proposal seems fine to me. Alternatively I would pitch something like `RHCOSMode`  or `StrictMode`, although I'm not strongly in favor of any of these 🙂